src/main/java/fr/insee/pearljam/batch/service/impl/MeshuggahServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+package fr.insee.pearljam.batch.service.impl;
+
+import fr.insee.pearljam.batch.communication.CommunicationTemplate;
+import fr.insee.pearljam.batch.exception.MissingCommunicationException;
+import fr.insee.pearljam.batch.exception.PublicationException;
+import fr.insee.pearljam.batch.exception.SynchronizationException;
+import fr.insee.pearljam.batch.service.KeycloakService;
+import fr.insee.pearljam.batch.service.MeshuggahService;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Collections;
+
+@Service
+@RequiredArgsConstructor
+public class MeshuggahServiceImpl implements MeshuggahService {
+    private static final Logger LOGGER = LogManager.getLogger(MeshuggahServiceImpl.class);
+
+    private final RestTemplate restTemplate;
```

Comment:
RestTemplate is deprecated. RestClient maybe ? the re's an implem example on meshuggah